Add HelmChart resource support and enhance HelmRelease handling

mouchar opened the pull request:
- Introduced HelmChart resource parsing and resolution in the HelmReleaseController.
- Updated HelmChart class to include additional fields: `chart` and `namespace`.
- Enhanced HelmVisitor to collect HelmChart resources and resolve them when referenced in HelmRelease.
- Updated related tests and snapshots to reflect changes in HelmChart structure and behavior.

## Notes
The primary motivation was the failing tests `test_build_hr[build-hr-single-cluster7]` and `test_build_hr_new[build-hr-single-cluster7]` which do not respect `spec.version` in tests/testdata/cluster7/flux/apps/ingress-nginx/chart.yaml and always install the latest available ingress-nginx.

I put the main HelmChart resolution logic to HelmReleaseController, which is probably not the best place, but I didn't want to make large refactors.

mouchar commented on flux_local/helm_controller/controller.py:
```diff
@@ -348,6 +353,60 @@ def on_object_added(resource_id: NamedResource, obj: BaseManifest) -> None:
+        )
+
+        helm_chart = self.store.get_object(chart_resource, HelmChart)
+        if not helm_chart:
```
No, helm-controller doesn't fail when HelmChart is not available. It marks HelmRelease as `Ready=false` and keeps trying to reconcile.

In scope of flux-local this behavior is not desirable, as the main purpose of flux-local is to allow local testing.
So I will raise HelmControllerException instead.

allenporter commented on flux_local/manifest.py:
```diff
@@ -232,6 +238,88 @@ def chart_name(self) -> str:
+    """The kind of the sourceRef."""
+
+    @classmethod
+    def parse_resource(cls, doc: dict[str, Any]) -> "HelmChartSource":
```
Please update to `parse_doc` to be consistent with the other data objects.

allenporter reviewed (changes requested):
Thank you again, looking forward to landing this. Couple questions/comments below.

mouchar commented:
I had to rework part of my code to make CI happy, but still one workflow fails on `Error: Resource not accessible by integration`, which is some type of permission issue from 3rd-party action.
I squashed all my fixups into a single commit, as they were becoming too messy.

I also removed a test I previously added - it actually didn't bring anything new. So I slightly changed `cluster7` data to show more complex conditions (when `HelmChart.namespace != HelmRelease.namespace`)

allenporter commented on flux_local/helm_controller/controller.py:
```diff
@@ -170,6 +172,9 @@ async def reconcile_helm_release(
         await self.wait_for_dependencies(helm_release)
 
+        # Resolve HelmChart resource if chartRef is used
+        await self._resolve_helm_chart(helm_release)
```
Do we still need the `wait_for_resource_ready` and `wait_for_resource_exists` inside `_resolve_helm_chart`? What is it waiting for given it already merged the helm chart by that point?

allenporter reviewed (approved):
Looks good, thanks for making those improvements. I have a couple last questions below.
